pkg/resources/task_state_upgraders.go

```diff
@@ -0,0 +1,49 @@
+package resources
+
+import (
+	"context"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+)
+
+func v098TaskStateUpgrader(ctx context.Context, rawState map[string]any, meta any) (map[string]any, error) {
+	if rawState == nil {
+		return rawState, nil
+	}
+
+	rawState["condition"] = rawState["when"]
+	rawState["started"] = booleanStringFromBool(rawState["enabled"].(bool))
```

Comment:
what if the previous one was false by default? in other boolean strings too

Reply:
mb, started is boolean, in this case we'll preserve the default, but after upgrade it's required so it has to be explicitly defined as not started (I'll mention it in the migration guide). In the other case we can only map this to false and if the user won't define it after we'll make a plan to change it to "default".